src/registrar/assets/js/get-gov.js

```diff
@@ -834,3 +834,436 @@ function hideDeletedForms() {
+    //fetch json of page of domain requests, given page # and sort
+    fetch(`/get-domain-requests-json/?page=${page}&sort_by=${sortBy}&order=${order}`)
+      .then(response => response.json())
+      .then(data => {
+        if (data.error) {
+          alert(data.error);
+          return;
+        }
+
+        // handle the display of proper messaging in the event that no domain requests exist in the list
+        if (data.domain_requests.length) {
+          domainRequestsWrapper.classList.remove('display-none');
+          noDomainRequestsWrapper.classList.add('display-none');
+        } else {
+          domainRequestsWrapper.classList.add('display-none');
+          noDomainRequestsWrapper.classList.remove('display-none');
+        }
+
+        // identify the DOM element where the domain request list will be inserted into the DOM
+        const tbody = document.querySelector('.dotgov-table__domain-requests tbody');
+        tbody.innerHTML = '';
+
+        // remove any existing modal elements from the DOM so they can be properly re-initialized
+        // after the DOM content changes and there are new delete modal buttons added
+        unloadModals();
+        data.domain_requests.forEach(request => {
+          const domainName = request.requested_domain ? request.requested_domain : `New domain request <span class="text-base font-body-xs">(${new Date(request.created_at).toLocaleString()} UTC)</span>`;
+          const submissionDate = request.submission_date ? new Date(request.submission_date).toLocaleDateString() : `<span class="text-base">Not submitted</span>`;
+          const actionUrl = (request.status === 'Started' || request.status === 'Withdrawn') ? `/domain-request/${request.id}/edit` : `/domain-request/${request.id}`;
+          const actionLabel = (request.status === 'Started' || request.status === 'Withdrawn') ? 'Edit' : 'Manage';
+          const deleteButton = request.is_deletable ? `
+            <a 
+              role="button" 
+              id="button-toggle-delete-domain-alert-${request.id}"
+              href="#toggle-delete-domain-alert-${request.id}"
+              class="usa-button--unstyled text-no-underline late-loading-modal-trigger"
+              aria-controls="toggle-delete-domain-alert-${request.id}"
+              data-open-modal
+            >
+              <svg class="usa-icon" aria-hidden="true" focusable="false" role="img" width="24">
+                <use xlink:href="/public/img/sprite.svg#delete"></use>
+              </svg> Delete <span class="usa-sr-only">${domainName}</span>
+            </a>` : '';
+
+          const row = document.createElement('tr');
+          row.innerHTML = `
+            <th scope="row" role="rowheader" data-label="Domain name">
+              ${domainName}
+            </th>
+            <td data-sort-value="${new Date(request.submission_date).getTime()}" data-label="Date submitted">
```

Comment:
0

Reply:
ANDI? Or screen reader? Not sure what the problem statement is but if it's a matter of an extra aria-label then no need for a ticket.

Reply:
ANDI does that thing which shows what a screen reader would read on an element but couldn't get it to read off the expiration date with it (or navigate to it at all really). Voiceover I also couldn't get it to read. Hopefully I wasn't being dumb with both. If you can add in an aria label for expiration dates that would be great. Otherwise, we can always do a follow on. Totally not really your problem here as it's unrelated to the main point of this ticket

Reply:
Punting